### ISSUE TITLE
Add Search Directory Manager for Native Libraries

### ISSUE THREAD

**Issue by HerpDerpinstine** [LavaGang/MelonLoader#153]
Native Libs only resolve from certain specific locations.
Something like the Assembly Resolve Manager has where it allows for specifying custom locations to search would be useful.

**Comment by HerpDerpinstine** [comment]
A wrapper using AddDllDirectory and RemoveDllDirectory might be viable.

**Comment by slxdy** [comment]
I dont think this is needed anymore, considering that native libraries are already loaded from UserLibs. Mods should not manage the resolution paths

**Comment by HerpDerpinstine** [comment]
> I dont think this is needed anymore, considering that native libraries are already loaded from UserLibs. Mods should not manage the resolution paths

This would also allow better management of the resolution pathing used by MelonLoader itself.
Cramming everything into the environment option isn't exactly the best way to deal with it.

**Comment by slxdy** [comment]
Ig that could work, so we don't have to preload everything and stuff is only loaded when it's actually needed.
